sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -287,7 +332,13 @@ private[sql] class RocksDBStateStoreProvider
       useColumnFamilies)
   }
 
-  @volatile private var encoder: RocksDBStateEncoder = _
+  private val encoderMapLock = new Object
```

Comment:
[nit] rename this to `columnFamilyEncoderMapLock` as it only applies to column families. Encoders for default column family are picked at State store init.

Reply:
Not sure I understand - we would still have to protect the same hashMap right ?

Reply:
Removed this